mailpoet/assets/js/src/segments/dynamic/templates/components/template_list_item.tsx

```diff
@@ -19,8 +21,36 @@ type TemplateListItemProps = {
 export function TemplateListItem({
   template,
 }: TemplateListItemProps): JSX.Element {
+  const segment: Segment = useSelect(
+    (select) => select(storeName).getSegment(),
+    [],
+  );
+
+  const { updateSegment, createFromTemplate } = useDispatch(storeName);
+
+  function handleSelectTemplate(segmentTemplate: SegmentTemplate): void {
+    segment.name = segmentTemplate.name;
+    segment.description = segmentTemplate.description;
+    segment.filters = segmentTemplate.filters;
+
+    if (segmentTemplate.filtersConnect) {
+      segment.filters_connect = segmentTemplate.filtersConnect;
+    }
+
+    updateSegment({
+      ...segment,
+    });
+    createFromTemplate();
```

Comment:
At first, I was a bit confused why we update a segment here and then call `createFromTemplate()`, but as I understand it, `createFromTemplate()` is pretty much just the final "save". In that case, we could also consider passing the `segmentTemplate` to `createFromTemplate(segmentTemplate)` and move the logic there. Not too important.

Reply:
I like this suggestion and I added a commit implementing it. Thanks.